pysph/solver/solver.py

```diff
@@ -198,6 +199,9 @@ def setup(self, particles, equations, nnps, kernel=None, fixed_h=False):
         )
 
         sep = '-'*70
+
+        if isinstance(equations, MultiStageEquations):
```

Comment:
A better approach would be to overload `__repr__` on the MultiStageEquations class as is done for the Group and Equation classes.

Reply:
Thanks! did the changes.